hydromt/raster.py

```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
     def _check_dimensions(self) -> None:
         """Validate the number and order of dimensions."""
         dims = (self.y_dim, self.x_dim)
-        da = self._obj[self.vars[0]] if isinstance(self._obj, xr.Dataset) else self._obj
-        extra_dims = [dim for dim in da.dims if dim not in dims]
+
+        def _check_dims(da: xr.DataArray, dims=dims) -> bool:
+            extra_dims = [dim for dim in da.dims if dim not in dims]
+            if len(extra_dims) == 1:
+                dims = tuple(extra_dims) + dims
+            elif len(extra_dims) > 1:
+                raise ValueError("Only 2D and 3D data arrays supported.")
+            return da.dims == dims, extra_dims
+
+        extra_dims = []
+        if isinstance(self._obj, xr.DataArray):
+            check, dim0 = _check_dims(self._obj)
+            extra_dims.extend(dim0)
+            if not check:
+                raise ValueError(
+                    f"Invalid dimension order ({self._obj.dims}). "
+                    f"You can use `obj.transpose({dims}) to reorder your dimensions."
+                )
+        else:
+            for var in self._obj.data_vars:
+                if not all([dim in self._obj[var].dims for dim in dims]):
+                    continue  # only check data variables with x and y dims
```

Comment:
Does this check data vars that _have_ x and y dims or that _only have_ x and y dims?

Reply:
That check happens when getting the x and y dims in the raster.x_dim and raster.y_dim properties.